src/vm/threads.cpp

```diff
@@ -7996,12 +7996,8 @@ NOINLINE void Thread::OnIncrementCountOverflow(UINT32 *threadLocalCount, UINT64
 
 UINT64 Thread::GetTotalCount(SIZE_T threadLocalCountOffset, UINT64 *overflowCount)
 {
-    CONTRACTL
-    {
-        NOTHROW;
-        MODE_ANY;
-    }
-    CONTRACTL_END;
+    WRAPPER_NO_CONTRACT;
```

Comment:
`WRAPPER_NO_CONTRACT` is meant to be used for trivial methods that wrap other method, and thus the contract can be easily inferred from the wrapped method. It is not the case here.

Should the contract for this be `STANDARD_METHOD_CONTRACT` ?

Reply:
I think the contract here could include `NOTHROW` but the GC contract from `ThreadSuspend::LockThreadStore` is a bit more complicated:

```c++
    CONTRACTL {
        NOTHROW;
        if ((GetThread() != NULL) && GetThread()->PreemptiveGCDisabled()) {GC_TRIGGERS;} else {DISABLED(GC_NOTRIGGER);}
    }
    CONTRACTL_END;
```

Since the code here doesn't have any specific contract requirements I figured I would let the callees determine the contract. I don't think it would be incorrect to say `NOTHROW` + `GC_TRIGGERS` if it had to be specific, I'll try that.